NMS-14447: Update Jira ticketing plugin doc (#5119)
OpenNMS/opennms: docs/modules/reference/pages/ticketing/ticketer/jira.adoc

----
+
If the ticketer properties file does not exist, create it manually.

. Configure the plugin by enabling and disabling options in `$\{OPENNMS_HOME}/etc/jira.properties`.
+
[options="header"]
[cols="1,3"]
|===
| Name
| Description

| jira.host
| Jira server URL.

| jira.username
| Username.

| jira.password
| An authentication token.
Must be configured in the Jira Cloud user settings.

| jira.project
| The Jira project's key.
Use the `jira:list-projects` Karaf command to see a list of project keys.

| jira.type
| The issue type ID to use when opening new issues.
Use the `jira:list-issue-types` Karaf command to see a list of issue type IDs (see https://confluence.atlassian.com/display/JIRA050/Finding+the+Id+for+Issue+Types[Finding the ID for issue types] in Atlassian's knowledge base for more information).

| jira.resolve
| Name of the transition to use when resolving issues.
The transition name is typically found on a button when looking at the ticket in Jira.

| jira.reopen
| Name of the transition to use when reopening issues.
The transition name is typically found on a button when looking at the ticket in Jira.

| jira.status.open

> | A comma-separated list of Jira status names under which a ticket should be considered 'Open.'

Reviewer: ```suggestion
| A comma-separated list of Jira status names that classify a ticket as "Open."
```

| jira.status.closed

> | A comma-separated list of Jira status names under which a ticket should be considered 'Closed.'

Reviewer: ```suggestion
| A comma-separated list of Jira status names that classify a ticket as "Closed."
```

| jira.status.cancelled

> | A comma-separated list of Jira status names under which a ticket should be considered 'Canceled.'

Reviewer: ```suggestion
| A comma-separated list of Jira status names that classify a ticket as "Canceled."
```

This is required to prevent the plugin from reading the issue type's metadata every time an issue is created.
A value of `0` disables the cache. +
The default value is `300000` (5 minutes).
|===

. To ensure the ticketer system starts automatically when the service starts, add `jira-troubleticketer` to the `$\{OPENNMS_HOME}/etc/featuresBoot.d/ticketer.boot` file.
If the file does not exist, create it manually.
. To start the ticketer system manually, run the following Karaf command:

 feature:install jira-troubleticketer

The plugin should now be ready to use.
Open the Karaf shell and run the `opennms:jira-verify` command to check your configuration.

NOTE: The `jira.verify` command will not work until the `jira.properties` file has been configured.

== Jira commands

The Jira ticketing plugin includes Karaf commands for setting up the plugin on the Karaf shell.
There are commands to list all available projects, versions, components, groups, issue types, and more.

To list all available commands, type `help | grep jira` in the Karaf shell.

Include `--help` after a command to see that command's documentation.

== Custom fields

> Besides the common fields (creator, create date, description, and subject), you also may also need to set custom fields.

Reviewer: ```suggestion
Besides the common fields (creator, create date, description, and subject), you also may need to set custom fields.
```

end
----

NOTE: Fields must be referenced by their field IDs.
Use the `opennms:jira-list-fields` Karaf command to determine a field's ID.
By default, only custom fields are shown.
You can show all fields by appending the `-s` option to the command.

You may also need to set default values (for example, the component, reporter, or assignee).

> The project key and issue types can also be redefined in the `jira.properties` file.

Reviewer: ```suggestion
You can also redefine the project key and issue types in the `jira.properties` file.
```

| Description

| any
| Any String.

| date

> | Any date in the _YYYY-MM-DD_ format.

Reviewer: ```suggestion
| Any date in YYYY-MM-DD format.
```

| datetime

> | Any date-time value in the ISO 8601 format (_YYYY-MM-DDThh:mm:ss.sTZD_).

Reviewer: ```suggestion
| Any date-time value in the ISO 8601 format (YYYY-MM-DDThh:mm:ss.sTZD).
```

custom datepicker              customfield_10101    true       date
----

. Defined components are `core`, `service`, and `web`.
. Defined versions are `1.0.0` and `1.0.1`.
. Defined issue types are `Bug` and `Task`.
. Defined priorities are `Major` and `Minor`.
. Defined projects are `NMS` and `HZN`.
. Defined options are `yes`, `no`, and `sometimes`.

> 

Reviewer 2: Not sure this section is needed as the following example re-states these.

Author: In that case, should we completely remove this example?

Reviewer: Should we keep the output example, but not the legend that explains everything?

Author: ```suggestion

```
